### PR TITLE
Resync cmd

### PR DESCRIPTION
**Pull request by pacmessica**
[notion](https://www.notion.so/figmentnetworks/b47c0f8b935741d8ac28717ddeea2038?v=5e4e1d3854094b788ae2b612329f27e8&p=bdbddafa3a2f4e0d86795090fa6cae67)

Adds `reindex` command which lets one reindex from specified for specified target (previously had to run `backfill` which required updating indexer_config and reindexing entire chain).

`go run main.go -config config.example.json -cmd indexer:reindex --start_height 3672055 --target_ids 4`

Summarizing balance events wasn't possible across multiple threads. We stopped indexing balance events ~2 weeks ago after the node upgrade. When we release the rewards fix and start indexing balance events again, we will have two processes running (a worker, and this reindex command). The worker will be indexing balance events from the chain head, and the reindex command will be indexing from 2 weeks ago. The `summarize` job in the worker wasn't handling summarizing the events from 2 weeks ago.

activity periods passed to [summarize](https://github.com/figment-networks/oasishub-indexer/blob/master/store/balance_events_store.go#L82) from look like this:
```
 period |          min           |          max
--------+------------------------+------------------------
      1 | 2021-01-01 00:00:00-05 | 2021-01-04 00:00:00-05
      2 | 2021-06-14 00:00:00-04 | 2021-06-14 00:00:00-04
(2 rows)
``` 

Previously, large gap for period 1 was handled like this `tx = tx.Or("time_bucket >= ? AND time_bucket < ?", activityPeriod.Max.Add(duration), activityPeriods[i+1].Min)`. This skips summarizing events on `2021-01-04 ` which are continuing to be collected, and instead captures only events between `2021-01-05` and  `2021-06-14`. Then when the indexer starts collecting for  `2021-01-05`, a balance summary entry is created capturing all indexed events for that date, however on subsequent summarize calls period 1 is now ` 1 | 2021-01-01 00:00:00-05 | 2021-01-05 00:00:00-05` so all further events indexed on `2021-01-05 `are ignored.
Changing the query to the following captures and summarizes  both events happening on `2021-01-05` and the following day `2021-01-06`:
```				
tx = tx.Or("time_bucket >= ? AND time_bucket < ?", activityPeriod.Max, activityPeriod.Max.Add(duration))
 tx = tx.Or("time_bucket >= ? AND time_bucket < ?", activityPeriod.Max.Add(duration), activityPeriod.Max.Add(duration*2))
```

todo:
- !! need to update [purge_worker_interval](https://github.com/figment-networks/oasishub-indexer/blob/master/config/config.go#L45) in deploy config (on provisioning?) to something big like "1000h" (~5 weeks). Last indexed balance event was on `2021-06-11`, and purge job should NOT delete events older than 24h (current setting) otherwise We'll be deleting events faster than we summarize them.

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -17,9 +20,32 @@ type Flags struct {
...
+	force              bool
+}
+
+type targetIds []int64
+
+func (i *targetIds) String() string {
+	return fmt.Sprint(*i)
+}
+
+func (i *targetIds) Set(value string) error {
```
Comment (lukanus) [inline cli/cli.go:37]: uuuu.... Fancy :P

File: store/balance_events_store.go [hunk 2]
```diff
@@ -103,7 +103,8 @@ func (s *balanceEventsStore) Summarize(interval types.SummaryInterval, activityP
 				if err != nil {
 					return nil, err
 				}
-				tx = tx.Or("time_bucket >= ? AND time_bucket < ?", activityPeriod.Max.Add(duration), activityPeriods[i+1].Min)
+				tx = tx.Or("time_bucket >= ? AND time_bucket < ?", activityPeriod.Max, activityPeriod.Max.Add(duration))
```
Comment (lukanus) [inline store/balance_events_store.go:106]: If I'm not mistaken it's  
```
tx = tx.Or("time_bucket >= ? AND time_bucket < ?", activityPeriod.Max, activityPeriod.Max.Add(duration*2))
```

**Comment by lukanus** [comment]
Other than this SQL thing, the rest looks good :)

File: store/balance_events_store.go (hunk 2 above)
Comment (pacmessica) [reply store/balance_events_store.go:106]: haha yes, if you value fewer lines of code, I suppose that's one way to do it 😛

**Pull request merged**